### PR TITLE
chore: popen using utf8 encoding

### LINKED ISSUE #209: UnicodeDecodeError - help will be appriciate!
_Exception in thread Thread-1 (save_and_display_stream):
Traceback (most recent call last):
  File "C:\Users\ziv\AppData\Local\Programs\Python\Python311\Lib\threading.py", line 1038, in _bootstrap_inner
    self.run()
  File "C:\Users\ziv\AppData\Local\Programs\Python\Python311\Lib\threading.py", line 975, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\ziv\AppData\Local\Programs\Python\Python311\Lib\site-packages\interpreter\code_interpreter.py", line
293, in save_and_display_stream
    for line in iter(stream.readline, ''):
  File "C:\Users\ziv\AppData\Local\Programs\Python\Python311\Lib\encodings\cp1255.py", line 23, in decode
    return codecs.charmap_decode(input,self.errors,decoding_table)[0]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
UnicodeDecodeError: 'charmap' codec can't decode byte 0x8e in position 3284: character maps to <undefined>_

I am a Windows User, running with Conda, on Python version 3.11.2
 should I change the encoding?

### PR DESCRIPTION
**Pull request by fatelei**
fix #209 using utf8 encoding instead of cp1255

### CODE REVIEW

**Comment by jordanbtucker** [comment]
Thanks for the PR. Do you think this will also fix #93 ?

**Comment by fatelei** [comment]
> Thanks for the PR. Do you think this will also fix #93 ?

i check it

**Review by anewt95 — APPROVED** [review]

File: interpreter/code_interpreter.py [hunk 1]
```diff
@@ -6,7 +6,6 @@
 import platform
 import time
 import ast
-import astor
 import sys
 import os
```
Comment (anewt95) [inline interpreter/code_interpreter.py:10]: []()

File: interpreter/code_interpreter.py [hunk 2]
```diff
@@ -76,14 +75,19 @@ def __init__(self, language, debug_mode):
   def start_process(self):
     # Get the start_cmd for the selected language
     start_cmd = language_map[self.language]["start_cmd"]
+    
+    # fix: https://github.com/KillianLucas/open-interpreter/issues/209
+    my_env = os.environ
```
Comment (anewt95) [inline interpreter/code_interpreter.py:80]: **[]()**

Comment (anewt95) [reply interpreter/code_interpreter.py:80]: image

**Review by anewt95 — APPROVED** [review]

**Comment by fatelei** [comment]
> Thanks for the PR. Do you think this will also fix #93 ?

i have reply this issue, how to resolve it, this mr only resolve subprocess popen encoding issue

**Review by anewt95 — APPROVED** [review]

**Review by anewt95 — CHANGES_REQUESTED** [review]
[]()

**Review by anewt95 — APPROVED** [review]

**Comment by ericrallen** [comment]
Hey there, @fatelei!

Sorry we didn’t get around to this one previously.

I’d love to get this updated to work with the refactored codebase and ready to merge in.

If you want help navigating the refactor, resolving merge conflicts, etc. let me know, or if you want to start with a fresh PR and avoid the complexities of the merge conflicts with the refactor, just ping me on the new PR.

**Comment by fatelei** [comment]
> Hey there, @fatelei!
> 
> Sorry we didn’t get around to this one previously.
> 
> I’d love to get this updated to work with the refactored codebase and ready to merge in.
> 
> If you want help navigating the refactor, resolving merge conflicts, etc. let me know, or if you want to start with a fresh PR and avoid the complexities of the merge conflicts with the refactor, just ping me on the new PR.

ok

**Comment by fatelei** [comment]
open a new one https://github.com/KillianLucas/open-interpreter/pull/742

**Pull request closed without merging**